src/vm/wren_value.h

```diff
@@ -114,6 +114,9 @@ typedef struct sObj
 
   // The next object in the linked list of all currently allocated objects.
   struct sObj* next;
+
+  // The next object in the GC gray set.
+  struct sObj* gcNext;
```

Comment:
It would be nice to avoid adding another field to every object if possible. The simplest solution is probably to store the gray set as an explicit collection. We can make an `Obj*` Buffer and add and remove gray objects to that.

What do you think?